### PR TITLE
[file_selector] Avoids using path_provider in web example app.

### PR DESCRIPTION
**Pull request by ditman**
This is a minor change to the `file_selector` example app to prevent it from using `path_provider` when running on the web (because `path_provider` isn't available on the web).

Fixes https://github.com/flutter/flutter/issues/130368

(PS: This should be exempt from publishing requirements, because the modified file is not shown in pub.dev, or anywhere else AFAIK.)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
> This should be exempt from publishing requirements

Not only is it, but the tooling knows that :)

(It does ask for a changelog for any example change in case that would be interesting to clients even though it's not on the pub.dev page, but I agree this one isn't.)

File: packages/file_selector/file_selector/example/lib/save_text_page.dart [hunk 1]
```diff
@@ -22,8 +22,9 @@ class SaveTextPage extends StatelessWidget {
     // This demonstrates using an initial directory for the prompt, which should
     // only be done in cases where the application can likely predict where the
     // file will be saved. In most cases, this parameter should not be provided.
-    final String initialDirectory =
-        (await getApplicationDocumentsDirectory()).path;
+    final String? initialDirectory = !kIsWeb
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector/example/lib/save_text_page.dart:25]: Optional nit: I find double negation harder to parse; I would prefer this as `kIsWeb ? null : ...`.

**Review by stuartmorgan — APPROVED** [review]
LGTM! Thanks for fixing this; I actually noticed this while adjusting something else in here recently, but forgot to follow up with a PR for it.

File: packages/file_selector/file_selector/example/lib/save_text_page.dart (hunk 1 above)
Comment (ditman) [reply packages/file_selector/file_selector/example/lib/save_text_page.dart:25]: Yes, I thought about this too! Let me fix it.

**Comment by ditman** [comment]
@stuartmorgan I removed the `-` root file in this PR.

**Pull request merged**